### PR TITLE
`treehouses redirect` (fixes #1493)

### LINKED ISSUE #1493: `treehouses hostnames`?
redirect internet hostnames strait to rpi

### PR DESCRIPTION
**Pull request by darthnoward**
fixes #1493 

```
root@treehouses:~# treehouses redirect add freebsd.org
root@treehouses:~# treehouses redirect list
freebsd.org
root@treehouses:~# ping -c 2 freebsd.org
PING freebsd.org (127.0.1.1) 56(84) bytes of data.
64 bytes from treehouses (127.0.1.1): icmp_seq=1 ttl=64 time=0.143 ms
64 bytes from treehouses (127.0.1.1): icmp_seq=2 ttl=64 time=0.162 ms
```
```
root@treehouses:~# treehouses redirect remove freebsd.org
root@treehouses:~# ping -c 2 freebsd.org
PING freebsd.org (96.47.72.84) 56(84) bytes of data.
64 bytes from wfe0.nyi.freebsd.org (96.47.72.84): icmp_seq=1 ttl=39 time=256 ms
64 bytes from wfe0.nyi.freebsd.org (96.47.72.84): icmp_seq=2 ttl=39 time=255 ms
```

### CODE REVIEW

**Comment by lmmrssa** [comment]
it only gives output when i am requesting from inside raspberrypi. If it request it from laptop or mobile being connected to raspberrypi it does not work

**Comment by lmmrssa** [comment]
remove redirect also does not work

**Comment by darthnoward** [comment]
now for this to work, `treehouses redirect start` will be needed if ip address of the rpi has changed.

File: modules/redirect.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+function check_ip {
+  ip="$(nmap -sn "$(ip route show | grep 'via' | awk '{print $3}')/24" 2>/dev/null |\
+    grep -i "$(</etc/hostname)" | awk '{print $6}' | sed -e 's/(//g' -e 's/)//g')"
+  if [ -z "$ip" ]; then
+    case "$(networkmode)" in
+      *wifi*)
+        ip="$(get_ipv4_ip wlan0)" ;;
+      *eth*)
+        ip="$(get_ipv4_ip eth0)" ;;
+      *ap* | *bridge*)
```
Comment (lmmrssa) [inline modules/redirect.sh:66]: ap mode uses eth0 only bridge seems to use ap0

**Comment by lmmrssa** [comment]
we might require script to auto update IP in DHCP system which right now `treehouses redirect start` does manually.
Also sometimes old dns seems to be cached so we might need to clear dns cache

**Pull request merged**